gctree/branching_processes.py

```diff
@@ -987,3 +964,230 @@ def f(x):
+                )
+            else:
+                raise
+        else:
+            raise
+    return (p, q)
+
+
+def clade_tree_to_ctree(
+    clade_tree,
+    root=None,
+    counts=None,
+    parsimony_score=None,
+    root_seq=None,
+    leaf_seqs=None,
+):
+    etetree = clade_tree.to_ete(
+        name_func=lambda n: n.attr["name"],
+        features=["sequence"],
+        feature_funcs={"abundance": lambda n: n.attr["abundance"]},
+    )
+
+    ctree = CollapsedTree(etetree)
+    # Here can do some validation on the tree:
+    # root name:
+    if root is not None:
+        if root not in ctree.tree.name:
+            raise RuntimeError(
+                f"collapsed tree should have root name '{root}' but has instead {ctree.tree.name}"
+            )
+    # counts:
+    if counts is not None:
+        for node in etetree.iter_leaves():
+            if node.name:
+                assert counts[node.name] == node.abundance
+        for node in ctree.tree.traverse():
+            if isinstance(node.name, tuple):
+                for name in node.name:
+                    if name in counts:
+                        assert node.abundance == counts[name]
+            else:
+                if node.name in counts:
+                    assert counts[node.name] == node.abundance
+                else:
+                    assert node.abundance == 0
+
+    # unnamed_seq issue:
+    for node in ctree.tree.traverse():
+        if node.name == "unnamed_seq":
+            raise RuntimeError("Some node names are missing")
+
+    # Parsimony:
+    if parsimony_score is not None:
+        if parsimony_score != sum([node.dist for node in ctree.tree.traverse()]):
+            raise RuntimeError(
+                "History DAG tree parsimony score does not match parsimony score provided"
+            )
+    # Root sequence:
+    if root_seq is not None:
+        if ctree.tree.sequence != root_seq:
+            raise RuntimeError(
+                "History DAG root node sequence does not match root sequence provided"
+            )
+    # Leaf names:
+    if leaf_seqs is not None:
+        # A dictionary of leaf sequences to leaf names
+        observed_set = {
+            node for node in ctree.tree.iter_descendants() if node.abundance > 0
+        }
+        for node in observed_set:
+            if not node.is_root() and leaf_seqs[node.sequence] != node.name:
+                raise RuntimeError("History DAG tree leaf names don't match sequences")
+        observed_seqs = {node.sequence for node in observed_set}
+        nonroot_observed_seqs = observed_seqs - {ctree.tree.sequence}
+        nonroot_leaf_seqs = set(leaf_seqs.keys()) - {ctree.tree.sequence}
+        if nonroot_leaf_seqs != nonroot_observed_seqs:
+            raise RuntimeError(
+                "Observed nonroot sequences in history DAG tree don't match "
+                "observed nonroot sequences passed in leaf_seqs."
+            )
+    return ctree
+
+
+def cmcounter_dagfuncs():
```

Comment:
Does this need to be a public function in the API that is shown in docs? If not, suggest renaming `_ cmcounter_dagfuncs`